src/phoenix/session/client.py

```diff
@@ -249,6 +264,142 @@ def log_traces(self, trace_dataset: TraceDataset, project_name: Optional[str] =
                 },
             ).raise_for_status()
 
+    def upload_dataset_csv(
+        self,
+        file: Union[str, Path],
+        input_keys: Iterable[str] = (),
+        output_keys: Iterable[str] = (),
+        metadata_keys: Iterable[str] = (),
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        action: Literal["create", "append"] = "create",
+    ) -> None:
+        """
+        Upload CSV file as dataset to the Phoenix server.
+
+        Args:
+            file (str | Path): Location of the csv file.
+            input_keys (Iterable[str]): List of column names used as input keys.
+                input_keys, output_keys, metadata_keys must be disjoint, and must
+                exist in CSV column headers.
+            output_keys (Iterable[str]): List of column names used as output keys.
+                input_keys, output_keys, metadata_keys must be disjoint, and must
+                exist in CSV column headers.
+            metadata_keys (Iterable[str]): List of column names used as metadata keys.
+                input_keys, output_keys, metadata_keys must be disjoint, and must
+                exist in CSV column headers.
+            name: (Optional[str]): Name of the dataset. Required if action=append.
```

Comment:
We should specify that a name will be automatically generated if none is provided.